packages/nimble-blazor/Tests/NimbleBlazor.Tests.Acceptance/Tests/WaferMapTests.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Playwright;
+using Xunit;
+
+namespace NimbleBlazor.Tests.Acceptance
+{
+    public class WaferMapTests : AcceptanceTestsBase
+    {
+        public WaferMapTests(PlaywrightFixture playwrightFixture, BlazorServerWebHostFixture blazorServerClassFixture)
+            : base(playwrightFixture, blazorServerClassFixture)
+        {
+        }
+
+        [Fact]
+        public async Task WaferMap_RendersCircleAsync()
+        {
+            await using (var pageWrapper = await NewPageForRouteAsync("WaferMapRendersCircle"))
+            {
+                var page = pageWrapper.Page;
+                var waferMap = page.Locator("nimble-wafer-map");
```

Comment:
Is there anything else that we can assert against that would indicate that all of the proper JS calls were executed? At the moment, it seems like the assertions being made here can be done in a unit test. The purpose of the acceptance tests is to verify that any code defined in `NimbleBlazor.lib.module.js` executed as expected.

Reply:
I think only the canvas painting can tell if the calls went through. I tried matching some bytes, but it doesn't seem to work. Do you have other ideas?

Reply:
It seems that [playwright js uses this library](https://playwright.dev/docs/test-snapshots#maxdiffpixels) for comparing screenshots. [This ](https://www.nuget.org/packages/StronglyTyped.PixelMatch.net)is supposed to be a c# port but it uses Bitmap which is only supported on windows

Reply:
Oh I forgot we have acceptance test infra in blazor! We should definitely make sure the code paths going through lib.Module are exercised like the getValidity method

I think we should avoid screenshot related approaches. I know playwright has the ability to evaluate JS logic on the page that we could use to probe the wafermap: https://playwright.dev/docs/evaluating

We would want that to be encapsulated in a page object. @msmithNI @jattasNI do yall have any insight on playwright page object and test patterns?

Reply:
The evaluated code returns different results based on where it's run. in Visual Studio 22 all tests pass with rgba 85 85 0 255, in the cli thei fail with mismatch of rgba 0000 and in the ci here its rgba 255 255 255 255

Reply:
>do yall have any insight on playwright page object and test patterns?

You can see the patterns we've used in SLE [in the Skyline End2EndTests](https://dev.azure.com/ni/DevCentral/_git/Skyline?_a=contents&version=GBmaster&path=/End2EndTests/tests/shared/page-objects)

Reply:
@munteannatan I see that this was moved into owners review and that the build is failing with your comment described here:

> The evaluated code returns different results based on where it's run. in Visual Studio 22 all tests pass with rgba 85 85 0 255, in the cli thei fail with mismatch of rgba 0000 and in the ci here its rgba 255 255 255 255

What is the expectation? Are you still investigating the behavior?

Reply:
I was not currently investigating it because I had a couple of days of medical leave, sorry. 

My idea is that the canvas may be rendered with different dimensions or that the headless rendering may affect the expected colors.

If you have an idea for a fix or something I could try to find out I will do it next week

Reply:
Thanks for the update and hope your doing well!